crates/cli/src/bin/wasm-bindgen-test-runner/headless.rs

```diff
@@ -121,8 +121,14 @@ pub fn run(server: &SocketAddr, shell: &Shell, timeout: u64) -> Result<(), Error
 
     // Visit our local server to open up the page that runs tests, and then get
     // some handles to objects on the page which we'll be scraping output from.
-    let url =
-        std::env::var("WASM_BINDGEN_TEST_ADDRESS").unwrap_or_else(|_| format!("http://{}", server));
+    let url = match std::env::var("WASM_BINDGEN_TEST_ADDRESS") {
+        Ok(u) => {
+            let mut url = Url::parse(&u)?;
+            let _ = url.set_port(url.port().or(Some(server.port())));
+            url.as_str().to_owned()
```

Comment:
```suggestion
            url.to_string()
```
This should work I believe.